pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -191,10 +191,14 @@ type CacheKeyGenerator interface {
 
 	// LabelValues should return a cache key for a label values request. The cache key does not need to contain the tenant ID.
 	// LabelValues can return ErrUnsupportedRequest, in which case the response won't be treated as an error, but the item will still not be cached.
+	// LabelValues should return a nil *GenericQueryCacheKey when it returns an error and
+	// should always return nil *GenericQueryCacheKey when the returned error is non-nil.
```

Comment:
I think you're saying the same thing twice.

Reply:
i am 🤦 
```suggestion
	// LabelValues should return a nil *GenericQueryCacheKey when it returns an error and
	// should always return non-nil *GenericQueryCacheKey when the returned error is nil.
```